quickwit/quickwit-search/src/leaf.rs

```diff
@@ -879,6 +877,130 @@ impl CanSplitDoBetter {
         }
     }
 
+    /// This function tries to detect upfront which splits contain the top n hits and convert other
+    /// split searches to count only searches. It also optimizes split order.
+    ///
+    /// Returns the search_requests with their split.
+    fn optimize(
+        &self,
+        request: Arc<SearchRequest>,
+        mut splits: Vec<SplitIdAndFooterOffsets>,
+    ) -> Result<Vec<(SplitIdAndFooterOffsets, SearchRequest)>, SearchError> {
```

Comment:
In most queries we end up not modifying the request. It feels silly to clone the object so many times when we in fact have an Arc at disposal.

```suggestion
    ) -> Result<Vec<(SplitIdAndFooterOffsets, Arc<SearchRequest>)>, SearchError> {
```

Reply:
Yes, but the behavior was the same before, but in the loop instead